packages/aws-cdk-lib/aws-apigateway/lib/restapi.ts

```diff
@@ -589,7 +589,7 @@ export abstract class RestApiBase extends Resource implements IRestApi {
     if (deploy) {
 
       this._latestDeployment = new Deployment(this, 'Deployment', {
-        description: props.description? props.description :'Automatically created by the RestApi construct',
+        description: props.deployOptions?.description ?? 'Automatically created by the RestApi construct',
```

Comment:
That make sense.
I found the issue for this imprementation. And it say this imprementation is to be able to set the description per deployment.
https://github.com/aws/aws-cdk/issues/20934

I think this PR make the imprementation close to customer thinking.

Reply:
This is good, but please consider backwards compatibility. This will change the description of all deployments that don't have an explicit description today. This bug has been in here for ~4 years and people have so far been fine with the current behavior.

A safer implementation seems to me:

```suggestion
  description: props.deployOptions?.description ?? props.description ?? 'Automatically created by the RestApi construct',
```